vars/infra.groovy

```diff
@@ -1,5 +1,13 @@
 #!/usr/bin/env groovy
 
+Boolean isRunsOnJenkinsAzure() {
```

Comment:
>  The `buildPipeline()` logic is specific to Azure, so I concluded it's better to have "Azure" in the method name

Assuming you mean `buildPlugin()`, then yes, it has behavior specific to Azure, but that's limited to Jenkins project infra. This method becomes public API for the pipeline library. Notably, we can update the Pipeline library if we ever change project hosting as needed, without this method having to return `false` unconditionally (because there's no Jenkins Azure infra anymore).